MapboxNavigation/CarPlayManager.swift

```diff
@@ -278,43 +278,54 @@ extension CarPlayManager: CPListTemplateDelegate {
             let location = placemark.location {
             let destinationWaypoint = Waypoint(location: location)
             interfaceController?.popTemplate(animated: false)
-            calculateRouteAndStart(to: destinationWaypoint, completionHandler: completionHandler)
+            preview(waypoints: [destinationWaypoint], completionHandler: completionHandler)
             return
         }
         #endif
         
         // Selected a favorite? or any item with a waypoint.
         if let userInfo = item.userInfo as? [String: Any],
             let waypoint = userInfo[CarPlayManager.CarPlayWaypointKey] as? Waypoint {
-            calculateRouteAndStart(to: waypoint, completionHandler: completionHandler)
+            preview(waypoints: [waypoint], completionHandler: completionHandler)
             return
         }
         
         completionHandler()
     }
     
-    public func calculateRouteAndStart(from fromWaypoint: Waypoint? = nil, to toWaypoint: Waypoint, completionHandler: @escaping () -> Void) {
+    public func preview(waypoints: [Waypoint], completionHandler: @escaping CompletionHandler) {
+        var waypoints = waypoints
         
-        guard let rootViewController = self.carWindow?.rootViewController as? CarPlayMapViewController,
-            let userLocation = rootViewController.mapView.userLocation,
-            let location = userLocation.location else {
-                completionHandler()
-                return
+        if waypoints.count == 1 {
+            guard let rootViewController = self.carWindow?.rootViewController as? CarPlayMapViewController,
+                let userLocation = rootViewController.mapView.userLocation,
+                let location = userLocation.location else {
+                    completionHandler()
+                    return
+            }
+            let name = NSLocalizedString("CARPLAY_CURRENT_LOCATION", bundle: .mapboxNavigation, value: "Current Location", comment: "Name of the waypoint associated with the current location")
```

Comment:
I think this method should either always add this current location waypoint or never add it. As it is, if the caller specifies two waypoints, do we know whether the caller intended the first waypoint to be the origin or an intermediate destination?